### ISSUE TITLE
Blocks Navigation menu: no semantic information for nested blocks

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#11713]
When the Blocks Navigation menu displays nested blocks, it visually uses some indentation:

![blocks navigation menu nesting 10545](https://user-images.githubusercontent.com/1682452/48301581-88034600-e4f0-11e8-9bbd-324a7c01ba78.png)

While the visual indication is clear, there's no semantic information or any other kind of information assistive technologies can use to communicate the blocks nesting level. Blocks are announced as if they were in a flat list.

The nesting level is an important information that needs to be communicated to all users. I'm not sure the Menu component is suitable for this purpose. While [ARIA menus can have sub-menus](https://www.w3.org/TR/wai-aria-practices-1.1/examples/menubar/menubar-1/menubar-1.html), I;d argue the information this component conveys is closer to an ARIA tree.

On the other hand, an ARIA tree is complex to interact with. Considering the menu items should be links as proposed in #11711 I'd recommend to keep things simple and just use a navigation menu that has the ability to render nested unordered lists when necessary. The item "checked" state could be communicated simply using some screen reader text. Nested lists give us the information we need for free:

Simple example: https://www.w3.org/WAI/tutorials/page-structure/content/#nested-lists

![screenshot 134](https://user-images.githubusercontent.com/1682452/48301784-a74fa280-e4f3-11e8-93ce-e3e622134444.png)

![screenshot 2018-11-10 at 14 17 20](https://user-images.githubusercontent.com/1682452/48301785-a7e83900-e4f3-11e8-93b3-fac52b622314.png)
